ruma-federation-api/src/query/get_custom_information/v1.rs

```diff
@@ -3,24 +3,26 @@
 use ruma_api::ruma_api;
 use serde_json::Value as JsonValue;
 
+use std::collections::BTreeMap;
+
 ruma_api! {
     metadata: {
         description: "Performs a single query request on the receiving homeserver. The query string arguments are dependent on which type of query is being made.",
         method: GET,
         name: "custom",
-        path: "/_matrix/federation/v1/query/:kind",
+        path: "/_matrix/federation/v1/query/:query_type",
         rate_limited: false,
         authentication: AccessToken,
     }
 
     request: {
         /// The type of query to make.
         #[ruma_api(path)]
-        pub kind: &'a str,
+        pub query_type: &'a str,
 
         /// The body of the query
-        #[ruma_api(query)]
-        pub body: &'a str,
+        #[ruma_api(query_map)]
+        pub body: BTreeMap<String, String>,
```

Comment:
This should probably be named sth. else. `body` imples request body, when this is really the query parameters. Maybe just `params` :shrug:

Reply:
Maybe `query`?

Reply:
I'd prefer `params` or `query_params`.

Reply:
@agraven friendly reminder that this should still be renamed, but otherwise this PR is ready :)